astropy/cosmology/core.py

```diff
@@ -517,7 +517,9 @@ def is_flat(self):
     def nonflat(self: _FlatCosmoT) -> _CosmoT:
         """Return the equivalent non-flat-class instance of this cosmology."""
 
-    def clone(self, *, meta: Mapping | None = None, to_nonflat: bool = False, **kwargs):
+    def clone(
+        self, *, meta: Mapping | None = None, to_nonflat: bool = False, **kwargs
+    ) -> Cosmology:
```

Comment:
```suggestion
    def clone(
        self, *, meta: Mapping | None = None, to_nonflat: bool = False, **kwargs
    ) -> Self:
```

Reply:
I thought that `Self` might not be the correct return type, depending on the value of `to_nonflat`.

Reply:
It's true in detail but in general this is more useful. The most correct would be to use a typing overload.

Reply:
You know `cosmology` much better than I do, so if you think `Self` is a more useful annotation then I'll use that. I'd just like #15686 to be resolved before I push changes here.